internal/ansible/handler/logging_enqueue_owner.go

```diff
@@ -1,63 +1,218 @@
+		log.Error(nil, "Expected exactly 1 kind for OwnerType", "owner type", fmt.Sprintf("%T", e.ownerType), "kinds", kinds)
+		return err
+	}
+	// Cache the Group and Kind for the OwnerType
+	e.groupKind = schema.GroupKind{Group: kinds[0].Group, Kind: kinds[0].Kind}
+	return nil
+}
+
+// getOwnerReconcileRequest looks at object and builds a map of reconcile.Request to reconcile
+// owners of object that match e.OwnerType.
+func (e *enqueueRequestForOwner) getOwnerReconcileRequest(object metav1.Object, result map[reconcile.Request]empty) {
+	// Iterate through the OwnerReferences looking for a match on Group and Kind against what was requested
+	// by the user
+	for _, ref := range e.getOwnersReferences(object) {
+		// Parse the Group out of the OwnerReference to compare it to what was parsed out of the requested OwnerType
+		refGV, err := schema.ParseGroupVersion(ref.APIVersion)
+		if err != nil {
+			log.Error(err, "Could not parse OwnerReference APIVersion",
+				"api version", ref.APIVersion)
+			return
+		}
+
+		// Compare the OwnerReference Group and Kind against the OwnerType Group and Kind specified by the user.
+		// If the two match, create a Request for the objected referred to by
+		// the OwnerReference.  Use the Name from the OwnerReference and the Namespace from the
+		// object in the event.
+		if ref.Kind == e.groupKind.Kind && refGV.Group == e.groupKind.Group {
+			// Match found - add a Request for the object referred to in the OwnerReference
+			request := reconcile.Request{NamespacedName: types.NamespacedName{
+				Name: ref.Name,
+			}}
+
+			// if owner is not namespaced then we should not set the namespace
+			mapping, err := e.mapper.RESTMapping(e.groupKind, refGV.Version)
+			if err != nil {
+				log.Error(err, "Could not retrieve rest mapping", "kind", e.groupKind)
+				return
+			}
+			if mapping.Scope.Name() != meta.RESTScopeNameRoot {
+				request.Namespace = object.GetNamespace()
+			}
+
+			result[request] = empty{}
+		}
+	}
+}
+
+// getOwnersReferences returns the OwnerReferences for an object as specified by the enqueueRequestForOwner
+// - if IsController is true: only take the Controller OwnerReference (if found)
+// - if IsController is false: take all OwnerReferences.
+func (e *enqueueRequestForOwner) getOwnersReferences(object metav1.Object) []metav1.OwnerReference {
+	if object == nil {
+		return nil
+	}
+
+	// If not filtered as Controller only, then use all the OwnerReferences
+	if !e.isController {
+		return object.GetOwnerReferences()
+	}
+	// If filtered to a Controller, only take the Controller OwnerReference
+	if ownerRef := metav1.GetControllerOf(object); ownerRef != nil {
+		return []metav1.OwnerReference{*ownerRef}
+	}
+	// No Controller OwnerReference found
+	return nil
+}
+
+// logEvent logs an event with the version, kind and name of the object and its owner.
+func (e *enqueueRequestForOwner) logEvent(eventType string, object, newObject client.Object) {
```

Comment:
Can we use comments to signal that most of this is directly from controller-runtime's implementation (this looks for the most part copy-pasted with this function added and a couple other tweaks?).

Reply:
@varshaprasad96 can correct me if I'm wrong, but I think the whole thing is copy-pasted from c-r and [this comment](https://github.com/operator-framework/ansible-operator-plugins/pull/37/files#diff-66988cc74efb2fc170ec2f070cefbb4f99ce53e7dea426cd143883275c0c760cR35-R37) referred to the whole thing?

Reply:
I did see that comment, but didn't put two and two together until I was about halfway through looking at the code and then it clicked. Would have saved me a bit of time if it was explicitly stated as part of this comment that this logic was pulled directly from controller-runtime and maybe included a link to the code in the controller-runtime repository, but at the end of the day not a huge deal to me. Just something I thought was worth calling out :)

Reply:
Okay, yeah, lemme try to clarify that better at the top as I think it's all from c-r. https://github.com/operator-framework/ansible-operator-plugins/pull/37/files#diff-66988cc74efb2fc170ec2f070cefbb4f99ce53e7dea426cd143883275c0c760cR35-R40 wdyt?

Reply:
Yes, so this from the c-r, except for probably 4-5 lines which add logging. It seems wrong to use copy their code, and change the header when majority of it is from the other repo. I'm not expert with licensing reqs, but it seemed reasonable to maintain the header.